camel-core/src/main/java/org/apache/camel/impl/DefaultExchange.java

```diff
@@ -109,9 +109,9 @@ public Exchange copy(boolean safeCopy) {
             exchange.getIn().setFault(getIn().isFault());
             if (getIn().hasHeaders()) {
                 exchange.getIn().setHeaders(safeCopyHeaders(getIn().getHeaders()));
-                // just copy the attachments here
-                exchange.getIn().copyAttachments(getIn());
             }
+            // just copy the attachments here
```

Comment:
As attachments are very very seldom in use, then this has negative performance impact. You should check via `hasAttachments` if any copying is needed.

Reply:
Okay, should I apply this change to the out message as well?

Reply:
Yeah that would be good